pyteal/ast/ecdsa.py

```diff
@@ -0,0 +1,164 @@
+from enum import Enum
+from typing import Tuple, TYPE_CHECKING, cast
+
+from pyteal.ast import Expr, MultiValue
+from pyteal.errors import TealTypeError, verifyTealVersion
+from pyteal.ir import Op, TealBlock, TealOp
+from pyteal.types import TealType, require_type
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+EcdsaPubkey = [TealType.bytes, TealType.bytes]
+
+
+class EcdsaCurve(Enum):
+    """Enum representing an elliptic curve specification used in ECDSA."""
+
+    Secp256k1 = 0
+
+
+class EcdsaVerifyExpr(Expr):
+    """Implements basic ECDSA verification functionality
+
+    This class shouldn't be directly used in a PyTeal program. The EcdsaVerify function
+    should be used instead."""
+
+    def __init__(
+        self,
+        curve: EcdsaCurve,
+        data: Expr,
+        sigA: Expr,
+        sigB: Expr,
+        pkX: Expr,
+        pkY: Expr,
+    ):
+        super().__init__()
+        require_type(data, TealType.bytes)
+        require_type(sigA, TealType.bytes)
+        require_type(sigB, TealType.bytes)
+        require_type(pkX, TealType.bytes)
+        require_type(pkY, TealType.bytes)
+
+        self.op = Op.ecdsa_verify
+        self.curve = curve
+        self.args = [data, sigA, sigB, pkX, pkY]
+
+    def __teal__(self, options: "CompileOptions"):
+        verifyTealVersion(
+            self.op.min_version,
+            options.version,
+            "TEAL version too low to use op {}".format(self.op),
+        )
+
+        return TealBlock.FromOp(
+            options, TealOp(self, self.op, cast(int, self.curve)), *self.args
```

Comment:
I have doubts about this curve conversion, since Python enums behave a bit strangely. I'd recommend making the enum inherit from `IntEnum`, then you can do `int(self.curve)` here

Reply:
You're right, my original enum broke during assemble. This wasn't caught by the tests because I was comparing the compilation result prior to assembly. I'll expand the tests as well as correct this problem.